src/Chainweb/Rosetta/RestAPI.hs

```diff
@@ -80,14 +80,17 @@ type RosettaApi_ =
 
 type RosettaApi (v :: ChainwebVersionT) = 'ChainwebEndpoint v :> Reassoc RosettaApi_
 
+-- TODO: Investigate if Rosetta Erros can be dynamic
 data RosettaFailure
     = RosettaChainUnspecified
     | RosettaInvalidChain
     | RosettaMempoolBadTx
     | RosettaUnparsableTx
     | RosettaInvalidTx
     | RosettaBadNetwork
+    deriving (Enum, Bounded)
```

Comment:
was the motivation for making `RosettaFailure` single constructors to make enum/bounded derivations simpler?

Reply:
Yes :)

I think we should make the messages "total" in this way. We also don't really need to tell them the exact string values that caused the errors. They should know that already, since they were the ones that made the request.

Reply:
gotcha! I was thinking this as well as I was writing the enum. As long as the error message is specific enough, they should be able to debug it.